frontend/src/component/feature/FeatureView/FeatureOverview/NewFeatureOverviewEnvironment/NewFeatureOverviewEnvironment.tsx

```diff
@@ -52,62 +52,75 @@ const StyledHeaderTitle = styled('span')(({ theme }) => ({
 }));
 
 interface INewFeatureOverviewEnvironmentProps {
-    environmentId: string;
+    hiddenEnvironments: string[];
 }
 
-export const NewFeatureOverviewEnvironment = ({
-    environmentId,
+export const FeatureOverviewEnvironment = ({
+    hiddenEnvironments,
 }: INewFeatureOverviewEnvironmentProps) => {
     const projectId = useRequiredPathParam('projectId');
     const featureId = useRequiredPathParam('featureId');
     const { metrics } = useFeatureMetrics(projectId, featureId);
     const { feature } = useFeature(projectId, featureId);
 
-    const featureMetrics = getFeatureMetrics(feature?.environments, metrics);
-    const environmentMetric = featureMetrics.find(
-        ({ environment }) => environment === environmentId,
-    );
-    const featureEnvironment = feature?.environments.find(
-        ({ name }) => name === environmentId,
-    );
+    const environments =
+        feature?.environments.filter(
+            ({ name }) => !hiddenEnvironments.includes(name),
+        ) || [];
 
-    if (!featureEnvironment)
+    if (!environments || environments.length === 0) {
         return (
             <StyledFeatureOverviewEnvironment className='skeleton'>
                 <Box sx={{ height: '400px' }} />
             </StyledFeatureOverviewEnvironment>
         );
+    }
 
-    return (
-        <StyledFeatureOverviewEnvironment>
-            <StyledHeader data-loading>
-                <StyledHeaderToggleContainer>
-                    <FeatureOverviewEnvironmentToggle
-                        environment={featureEnvironment}
+    return environments.map(({ name: environmentId }) => {
+        const featureMetrics = getFeatureMetrics(
+            feature?.environments,
+            metrics,
+        );
```

Comment:
Probably not super important, but can we move this function call to outside the loop? It doesn't appear to be dependent on the environment id, so the response should be the same for each environment. I'm guessing it doesn't do any very computation, but in case it fires off some requests or does something silly 💁🏼

Reply:
I'll refactor it later